README.md

```diff
@@ -4,24 +4,58 @@
 
 # Django app for City of Helsinki user infrastructure
 
-django-helusers is your friendly app for bolting authentication into Django projects for City of Helsinki. It provides the following functionalities:
+Django-helusers is your friendly app for bolting authentication into Django projects for City of Helsinki. Authentication schemes are based on [OAuth2](https://oauth.net/2/) and [OpenID Connect (OIDC)](https://openid.net/connect/).
 
-* baseline User model
+A baseline `User` model is provided that can be used with the various authentication use cases that are supported. The model supports mapping from AD groups to Django groups based on the authentication data.
+
+Additionally there are **optional** functionalities that can be used as needed.
+
+Functionalities for server needing (API) access token verification:
+
+* For servers using Django REST Framework
+* For servers not using Django REST Framework
+
+Functionalities for server needing support for OAuth2 or OIDC based authentication (e.g. into the Django admin interface):
+
+* augmented login template for Django admin, allowing Tunnistamo login
```

Comment:
And this line can then be removed entirely.

Reply:
Changing this exactly as suggested produces this:

```
Functionality for the normal Django login form, that allows logging in using OAuth2/OIDC
in addition to username and password.
This can be used e.g. to allow OIDC login to Django admin:

* Authentication against Tunnistamo, an OIDC service for authenticating against multiple backends
```

This looks a bit cumbersome. How exactly would you change this?

Reply:
I see now, better have the symmetry (sentences, with lists below). So like this:
Functionalities for server needing to authenticate against OIDC or OAuth2 server
* support Django session login against OIDC or OAuth2 server, including Helsinki Tunnistus service and Azure AD.
* augmented login template for Django admin, adding OIDC/OAuth2 login button

Reply:
Is "Helsinki Tunnistus" a thing?

Reply:
It is the best name we've been able to come up with. Maybe @tkutinla could confirm the nomenclature?

Reply:
Is the "Helsinki Tunnistus" part important, if the mentioned OIDC/OAuth2 support covers all such standard servers anyway?

Reply:
Mostly to make it clear to folks, that might not know the standards.